File: alwin-web-grapescode/src/app/holiday/holiday.service.ts
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {AlwinHttpService} from '../shared/authentication/alwin-http.service';
import {Holiday} from './calendar/util/holiday';
import {HolidayInput} from './calendar/util/holiday-input';

/**
 * Serwisu dostępu do usług REST dla dni wolnych
 */
@Injectable()
export class HolidayService {

  constructor(private http: AlwinHttpService) {
  }

  /**
   * Pobiera dni wolne w podanym roku
   *
   * @param {number} year - rok
   * @returns {Observable<Holiday[]>} lista dni wolnych
   */
  getHolidays(year: number): Observable<Holiday[]> {
    return this.http.get<Holiday[]>(`holidays/${year}`);
  }

  /**
   * Usuwa wybrany dzień wolny
   * @param {number} holidayId  - identyfikator dnia wolnego
   * @returns {Observable<HttpResponse<Object>>} - odpowiedź http z poprawnym kodem usunięcia 204
   */
  deleteHoliday(holidayId: number) {
    return this.http.delete(`holidays/${holidayId}`);
  }

  /**
   * Tworzy nowy dzień wolny
   *
   * @param {Holiday} holiday - szczegóły dnia wolnego
   * @returns {Observable<HttpResponse<Object>>} - odpowiedź http z poprawnym kodem utworzenia 201
   */
  createHoliday(holiday: Holiday): any {
    return this.http.postObserve('holidays/', holiday);
  }

  /**
   * Aktualizuje dzień wolny o podanym identyfikatorze
   *
   * @param {number} holidayId  - identyfikator dnia wolnego
   * @param {HolidayInput} holiday - szczegóły dnia wolnego
   * @returns {Observable<HttpResponse<Object>>} - odpowiedź http z poprawnym kodem aktualizacji 202
   */
  updateHoliday(holidayId: number, holiday: HolidayInput): any {
    return this.http.patch(`holidays/${holidayId}`, holiday);
  }

}

File: alwin-middleware-grapescode/alwin-core-api/src/main/java/com/codersteam/alwin/core/api/service/holiday/HolidayService.java
package com.codersteam.alwin.core.api.service.holiday;

import com.codersteam.alwin.core.api.model.holiday.HolidayDto;
import com.codersteam.alwin.core.api.model.holiday.HolidayInputDto;

import javax.ejb.Local;
import java.util.List;

/**
 * Serwis do obsługi dni wolnych
 *
 * @author Michal Horowic
 */
@Local
public interface HolidayService {

    /**
     * Pobiera listę dni wolnych dla danego dnia, w przypadku przekazania identyfikatora użytkownika zwraca tylko jego dni wolne
     *
     * @param day    - dzień do sprawdzenia dni wolnych
     * @param month  - miesiąc dla dnia do sprawdzenia dni wolnych
     * @param year   - rok dla dnia do sprawdzenia dni wolnych
     * @param userId - identyfikator użytkownika
     * @return lista dni wolnych dla danego dnia
     */
    List<HolidayDto> findAllHolidaysPerDay(final int day, final int month, final int year, final Long userId);

    /**
     * Pobiera listę dni wolnych dla danego miesiąca, w przypadku przekazania identyfikatora użytkownika zwraca tylko jego dni wolne
     *
     * @param month  - miesiąc do sprawdzenia dni wolnych
     * @param year   - rok dla miesiąc do sprawdzenia dni wolnych
     * @param userId - identyfikator użytkownika
     * @return lista dni wolnych dla danego miesiąca
     */
    List<HolidayDto> findAllHolidaysPerMonth(final int month, final int year, final Long userId);

    /**
     * Pobiera listę dni wolnych dla danego roku, w przypadku przekazania identyfikatora użytkownika zwraca tylko jego dni wolne
     *
     * @param year   - rok do sprawdzenia dni wolnych
     * @param userId - identyfikator użytkownika
     * @return lista dni wolnych dla danego roku
     */
    List<HolidayDto> findAllHolidaysPerYear(final int year, final Long userId);

    /**
     * Tworzy nowy dzień wolny
     *
     * @param holiday - szczegóły dnia wolnego
     */
    void createNewHoliday(final HolidayInputDto holiday);

    /**
     * Aktualizuje istniejący dzień wolny
     *
     * @param holidayId - identyfikator dnia wolnego
     * @param holiday   - szczegóły dnia wolnego
     */
    void updateHoliday(final long holidayId, final HolidayInputDto holiday);

    /**
     * Usuwa istniejący dzień wolny
     *
     * @param holidayId - identyfikator dnia wolnego
     */
    void deleteHoliday(final long holidayId);

    /**
     * Sprawdza czy dzień wolny o podanym identyfikatorze istnieje
     *
     * @param holidayId - identyfikator dnia wolnego
     * @return true jeżeli dzień wolny istnieje, false w przeciwnym razie
     */
    boolean checkIfHolidayExists(final long holidayId);
}
